### PR TITLE
[libc++] Use explicit #include instead of transitive #include

### PR DESCRIPTION
**Pull request by winner245**
This benchmark test currently uses `std::unique_ptr` without explicitly `#include <memory>`. I think we should not rely on transitive inclusion.

### CODE REVIEW

File: libcxx/test/benchmarks/vector_operations.bench.cpp [hunk 1]
```diff
@@ -14,6 +14,7 @@
 #include <deque>
 #include <functional>
 #include <vector>
+#include <memory>
```
Comment (frederick-vs-ja) [inline libcxx/test/benchmarks/vector_operations.bench.cpp:17]: Currently this file also uses `std::string` without explictly including `<string>`. Perhaps we should add `#include <string>` together.

Comment (winner245) [reply libcxx/test/benchmarks/vector_operations.bench.cpp:17]: Thank you for your reminder. I'll also add that.

File: libcxx/test/benchmarks/vector_operations.bench.cpp [hunk 2]
```diff
@@ -14,6 +14,8 @@
 #include <deque>
 #include <functional>
 #include <vector>
+#include <string>
+#include <memory>
```
Comment (philnik777) [inline libcxx/test/benchmarks/vector_operations.bench.cpp:18]:
```suggestion
#include <deque>
#include <functional>
#include <memory>
#include <string>
#include <vector>
```

Comment (winner245) [reply libcxx/test/benchmarks/vector_operations.bench.cpp]: Thanks. I've adjusted the order of headers as per your suggestion.

**Review by philnik777 — APPROVED** [review]

**Pull request merged**